packages/@tailwindcss-upgrade/src/index.ts

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env node
+
+import fastGlob from 'fast-glob'
+import path from 'node:path'
+import { help } from './commands/help'
+import { migrate } from './migrate'
+import { args, type Arg } from './utils/args'
+import { isRepoDirty } from './utils/git'
+import { eprintln, error, header, highlight, info, success } from './utils/renderer'
+
+const options = {
+  '--help': { type: 'boolean', description: 'Display usage information', alias: '-h' },
+  '--force': { type: 'boolean', description: 'Force the migration', alias: '-f' },
+  '--version': { type: 'boolean', description: 'Display the version number', alias: '-v' },
+} satisfies Arg
+const flags = args(options)
+
+if (flags['--help']) {
+  help({
+    usage: ['npx @tailwindcss/upgrade'],
+    options,
+  })
+  process.exit(0)
+}
+
+async function run() {
+  eprintln(header())
+  eprintln()
+
+  if (!flags['--force']) {
+    if (isRepoDirty()) {
+      error('Git directory is not clean. Please stash or commit your changes before migrating.')
+      info(
+        `You may use the ${highlight('--force')} flag to silence this warning and perform the migration.`,
+      )
+      process.exit(1)
+    }
+  }
+
+  // Use provided files
+  let files = flags._.map((file) => path.resolve(process.cwd(), file))
+
+  // Discover CSS files in case no files were provided
+  if (files.length === 0) {
+    info(
+      'No files provided. Searching for CSS files in the current directory and its subdirectories…',
+    )
+
+    files = await fastGlob(['**/*.css'], {
+      absolute: true,
+      ignore: ['**/node_modules', '**/vendor'],
```

Comment:
Wonder if we should instead use the `.gitignore` info here, I remember that we had a few people who had `vendor` folders when we added the content-path-warning and we eventually had to revert it.

Reply:
Hmm yeah, let's see what I can find to solve this!

Reply:
Found globby https://github.com/sindresorhus/globby, based on fast-glob and has support for this out of the box

Reply:
:chef-kiss: